packages/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -596,7 +596,7 @@ async function resolveDependency (
       preferredVersions: options.preferredVersions,
       preferWorkspacePackages: ctx.preferWorkspacePackages,
       projectDir: options.currentDepth > 0 ? ctx.lockfileDir : ctx.prefix,
-      registry: wantedDependency.alias && pickRegistryForPackage(ctx.registries, wantedDependency.alias) || ctx.registries.default,
+      registry: wantedDependency.alias && pickRegistryForPackage(ctx.registries, wantedDependency.alias, wantedDependency.pref) || ctx.registries.default,
```

Comment:
Search for other places in the repository where `pickRegistryForPackage` is used and update it if pref is also availabe.